### ISSUE TITLE
Invalid expiration date value

### ISSUE THREAD

**Issue by tstavinoha** [relayrides/pushy#237]
Hello, we've encountered a bug regarding the expiration date value.
For registrations which expired today, pushy returned the expiration date of Tue Jan 01 04:20:19 UTC **48075** (yup, that's a year :) )

Now if we take that date, get its millis using getTime, divide it by 1000 and create a new Date using that value and voila - correct date is here! :)

I would assume that APNS returns the timestamp in nanos, however i was not able to find that piece of information in their documentation. Other possibility is that APNS returns milliseconds but pushy interprets it as seconds and multiplies it by 1000.. Not sure what's going on, but help would be appreciated :)

**Comment by jchambers** [comment]
Huh. So according to the ["Notification API" docs](https://developer.apple.com/library/ios/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/Chapters/APNsProviderAPI.html#//apple_ref/doc/uid/TP40008194-CH101-SW15), the `apns-expiration` header (which is what we send TO the APNs server) is:

> A UNIX epoch date expressed in seconds (UTC).

Meanwhile, the `timestamp` field of the payload FROM the server for a rejected notification has this description:

> If the value in the :status header is 410, the value of this key is the last time at which APNs confirmed that the device token was no longer valid for the topic.

I guess I had just assumed that both timestamps would be in the same units, but they never really say what the units are for the `timestamp` field, so it's possible that `apns-expiration` is in seconds, but `timestamp` is in milliseconds. That would be weird, but also not the weirdest thing related to APNs ;)

I'll try to confirm this on my own, but if you just happened to have your logging level set to `TRACE` when this happened, could you please post the original rejection payload from the APNs server here? The line in question would look something like this:

```
Received data from APNs gateway on stream 74893: { json payload here}
```

Thanks!

**Comment by jchambers** [comment]
Yup:

```
{"reason":"Unregistered","timestamp":1454948015990}
```

I fixed this in Pushy in bdd5faa. I'm concerned the units for the `apns-expiration` header may also be wrong, but we have no way of knowing how the server is interpreting the values we send there.

Additionally, I've reported the documentation issue as [rdar://24548417](https://openradar.appspot.com/radar?id=5056110025244672).
